testing/time.ts

```diff
@@ -0,0 +1,436 @@
+  this: FakeDate,
+  value: string | number | Date,
+): void;
+function FakeDate(
+  this: FakeDate,
+  year: number,
+  month: number,
+  date?: number,
+  hours?: number,
+  minutes?: number,
+  seconds?: number,
+  ms?: number,
+): void;
+function FakeDate(
+  this: FakeDate | void,
+  // deno-lint-ignore no-explicit-any
+  ...args: any[]
+): string | void {
+  if (args.length === 0) args.push(FakeDate.now());
+  if (isFakeDate(this)) {
+    this.date = new _internals.Date(...(args as []));
+  } else {
+    return new _internals.Date(args[0]).toString();
+  }
+}
+
+FakeDate.parse = Date.parse;
+FakeDate.UTC = Date.UTC;
+FakeDate.now = () => time?.now ?? _internals.Date.now();
+Object.getOwnPropertyNames(Date.prototype).forEach((name: string) => {
+  const propName: keyof Date = name as keyof Date;
+  FakeDate.prototype[propName] = function (
+    this: FakeDate,
+    // deno-lint-ignore no-explicit-any
+    ...args: any[]
+    // deno-lint-ignore no-explicit-any
+  ): any {
+    // deno-lint-ignore no-explicit-any
+    return (this.date[propName] as (...args: any[]) => any).apply(
+      this.date,
+      args,
+    );
+  };
+});
+Object.getOwnPropertySymbols(Date.prototype).forEach((name: symbol) => {
+  const propName: keyof Date = name as unknown as keyof Date;
+  FakeDate.prototype[propName] = function (
+    this: FakeDate,
+    // deno-lint-ignore no-explicit-any
+    ...args: any[]
+    // deno-lint-ignore no-explicit-any
+  ): any {
+    // deno-lint-ignore no-explicit-any
+    return (this.date[propName] as (...args: any[]) => any).apply(
+      this.date,
+      args,
+    );
+  };
+});
+
+interface Timer {
+  id: number;
+  // deno-lint-ignore no-explicit-any
+  callback: (...args: any[]) => void;
+  delay: number;
+  args: unknown[];
+  due: number;
+  repeat: boolean;
+}
+
+export interface FakeTimeOptions {
+  /**
+   * The rate relative to real time at which fake time is updated.
+   * By default time only moves forward through calling tick or setting now.
+   * Set to 1 to have the fake time automatically tick foward at the same rate in milliseconds as real time.
+   */
+  advanceRate: number;
+  /**
+   * The frequency in milliseconds at which fake time is updated.
+   * If advanceRate is set, we will update the time every 10 milliseconds by default.
+   */
+  advanceFrequency?: number;
+}
+
+interface DueNode {
+  due: number;
+  timers: Timer[];
+}
+
+let time: FakeTime | undefined = undefined;
+
+function fakeSetTimeout(
+  // deno-lint-ignore no-explicit-any
+  callback: (...args: any[]) => void,
+  delay = 0,
+  // deno-lint-ignore no-explicit-any
+  ...args: any[]
+): number {
+  if (!time) throw new TimeError("no fake time");
+  return setTimer(callback, delay, args, false);
+}
+
+function fakeClearTimeout(id?: number): void {
+  if (!time) throw new TimeError("no fake time");
+  if (typeof id === "number" && dueNodes.has(id)) {
+    dueNodes.delete(id);
+  }
+}
+
+/** The fake time replacement for se */
```

Comment:
Seems like this comment was cut-off by mistake

Reply:
I'll look into what it was supposed to say tonight. My bad.

Reply:
This comment can just be deleted. The function is never exported and the fakeSetTimeout function doesn't have a comment above it either.

Reply:
I deleted this comment.